### DISCUSSION TITLE
Accessing Context API data in getServerSideProps

### DISCUSSION THREAD

**Discussion by ryanbahan** [vercel/next.js#14679] [Help]
Hey y'all! So I know that you can't use hooks like useContext inside of getServersideProps, but I'm curious if there is any feasible way to access data from my Context API on the server side (given that there are examples of accessing Redux in getInitialProps).

The issue I'm trying to solve is that on first page load, I get an initial payload from my fetch and store that data in Context. When navigating to other pages, there are instances where I don't need to perform another fetch because I have all the data necessary in Context. Currently, I'm just using a useContext hook on the client side. But I'd rather render that data on the server in some cases. 

Has anyone found a solution or workflow that gets around the issues described here? Would a better solution be to create an API in pages that handles this data, and make requests to/from that?

**Comment by ryanbahan** [comment]
Bumping this.

**Reply by TeoSlayer** [reply]
Having the same pronlem

**Reply by TeoSlayer** [reply]
Did you find anything in the meantime?

**Reply by CharterXavi** [reply]
Also curious about this. I'm new to Nextjs and figuring out where to do API calls and how to integrate ContextAPI and whatnot is confusing. Any good resources for architectural patterns like that while we're at it?

**Reply by TeoSlayer** [reply]
Hi, found a pretty handy solution. You can store the data as a token in a cookie and then fetch it in serversideprops.

**Reply by chloesun** [reply]
Facing the same issue, it would be nice and intuitive to use  `useContext` to access token in `getServerSideProps` to make authenticated requests

**Reply by treeindev** [reply]
`useContext` is a key element for components. I think `getServerSideProps` should expose access to read React context values. The cookie solution might work, but I still think they should define a more "React friendly" approach to it.

**Comment by jeni1616** [comment]
Guys, if you want to learn this, try integrating the next auth js ( https://next-auth.js.org/ ) in your project, it gives a good example of context API inside getServerSideProps

Quick example: https://next-auth.js.org/tutorials/securing-pages-and-api-routes
